### PR TITLE
Refactor/subject listing use new api - LESQ-730

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: {owa_music_year}

- remove query to legacy api on subject listing page
- update tests to use the 2023 fixture

1. Go to https://deploy-preview-2347--oak-web-application.netlify.thenational.academy

How to test
Go to:
Click on subject listing page for different keystages
You should see everything working as expected

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/pages/teachers/key-stages/[keyStageSlug]/subjects/index.tsx [hunk 1]
```diff
@@ -94,21 +94,37 @@ export const getStaticProps: GetStaticProps<
...
-      const isEyfs = keystage === "early-years-foundation-stage";
+      const keyStage = context.params?.keyStageSlug;
+
+      const isEyfs = keyStage === "early-years-foundation-stage";
 
-      const curriculumData2023 = await curriculumApi2023.subjectListingPage({
-        keyStageSlug: keystage,
+      const curriculumData = await curriculumApi2023.subjectListingPage({
+        keyStageSlug: keyStage,
         isLegacy: isEyfs,
```
Comment (JBR90) [inline src/pages/teachers/key-stages/[keyStageSlug]/subjects/index.tsx:103]: eyfs only exists in legacy

**Comment by JBR90** [comment]
image
@jack-skerrett-oak - there's a small MV change to fix the keysatge title here, which could happen before or after this PR is in ?

**Review by njblore — APPROVED** [review]

**Comment by jack-skerrett-oak** [comment]
> image @jack-skerrett-oak - there's a small MV change to fix the keysatge title here, which could happen before or after this PR is in ?

@JBR90 I think for this one I would be on the side of making the MV change... I imagine people will be annoyed if it changes to "KS", I know it feels like wasted effort but probably should be done prior to the PR merge

**Comment by JBR90** [comment]
@jack-skerrett-oak Sure, no problem

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
There seems to be an issue with legacy KS4 maths. It takes you to the programme listing page for new maths. Same for combined science KS4

**Comment by JBR90** [comment]
> There seems to be an issue with legacy KS4 maths. It takes you to the programme listing page for new maths. Same for combined science KS4

Looks like these two programmes are missing the legacy prefix on the database, i've flagged it with the other squads and have added fix for these programmes for now.

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**